api/opds.py

```diff
@@ -1722,7 +1723,17 @@ def single_item_feed(
             )
 
         _db = Session.object_session(item)
-        work = license_pool.work or license_pool.presentation_edition.work
+
+        # Sometimes the pool or work may be None
+        # In those cases we have to protect against the exceptions
+        try:
+            work = license_pool.work or license_pool.presentation_edition.work
+        except AttributeError:
+            return NOT_FOUND_ON_REMOTE
```

Comment:
Because we might return a problem detail, it would be good to update the docstring `:return:` of this method. While we're at it, it would be useful to add type hints, including the return types.